File: distros/O/OpenSocialX-Shindig-Crypter/sample/crypt.pl
#!/usr/bin/perl

use strict;
use warnings;
use MIME::Base64;
use Crypt::CBC;
use Digest::SHA;

my $str    = 'o=1&v=3&p=5';
my $cipher = Crypt::CBC->new(
    {
        'key'         => 'length16length16',
        'cipher'      => 'Rijndael',
        'iv'          => '1234567890abcdef',
        'literal_key' => 1,
        'padding'     => 'null',
        'header'      => 'none',
        keysize       => 128 / 8
    }
);
my $encrypted = $cipher->encrypt($str);
print "encrypted: " . encode_base64($encrypted) . "\n";
print "decrypted: " . $cipher->decrypt($encrypted) . "\n";

my $hmac = Digest::SHA::hmac_sha1( $encrypted, 'hmackey' );
print "hmac: " . encode_base64($hmac) . "\n";
print "total: " . encode_base64( $encrypted . $hmac ) . "\n";

1;

File: distros/O/OpenSocialX-Shindig-Crypter/sample/crypt.php
<?php

$str = 'o=1&v=3&p=5';
$encrypted = mcrypt_cbc(MCRYPT_RIJNDAEL_128, 'length16length16', $str, MCRYPT_ENCRYPT, '1234567890abcdef');
$decrypted = mcrypt_cbc(MCRYPT_RIJNDAEL_128, 'length16length16', $encrypted, MCRYPT_DECRYPT, '1234567890abcdef');
echo "encrypted: ".base64_encode($encrypted)."\n"; 
echo "decrypted: ".$decrypted."\n"; 

$blabla = hash_hmac('sha1', $encrypted, 'hmackey', 1);
echo "hmac: " . base64_encode($blabla) . "\n";
echo "total: " . base64_encode($encrypted . $blabla) . "\n";
